### PR TITLE
cdc: back pressure scan speed (#10142)

### PR DESCRIPTION
**Pull request by overvenus**
This PR cherry-pick #10142

----------------

### What problem does this PR solve?

There will be a series of PRs aims to fix the TiKV CDC OOM issue (too many pending send messages are buffered in memory).

To resolve the issue, we have two strategies,
1. Limit the speed of producing messages during the initial scan stage. #9948 
2. Set an upper bound of pending send messages. #9885 

The series of PRs adopts both strategies.

This is part of #9885, it backpressure scan speed by replacing an unbounded channel with a bounded channel.

Plan:
- [x] take1: cherry-pick speed limiter. #10108 
- [x] take2: set an upper bound of an internal channel used in CDC #10142 <-- we are here.
- [ ] take3: add memory quota to CDC and kill connection if it uses too much memory.
- [ ] take4: cherry-pick other bugfixes in #9885 

See full changes: https://github.com/tikv/tikv/compare/master...overvenus:cdc/flowcontrol/all

Cc #8168 #9996

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- Support back pressure CDC scan speed.

### CODE REVIEW

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -886,13 +905,8 @@ impl Initializer {
...
             debug!("cdc scan entries"; "len" => entries.len(), "region_id" => region_id);
             fail_point!("before_schedule_incremental_scan");
-            let scanned = Task::IncrementalScan {
-                region_id,
-                downstream_id,
-                entries,
-            };
-            if let Err(e) = self.sched.schedule(scanned) {
-                error!("cdc schedule cdc task failed"; "error" => ?e, "region_id" => region_id);
+            if let Err(e) = self.sink_scan_events(entries, done, require_barrier).await {
```
Comment (5kbpers) [inline components/cdc/src/endpoint.rs:908]: Please confirm ticdc checks `RequestId` in release-4.0 too.

**Review by 5kbpers — APPROVED** [review]

File: components/cdc/src/endpoint.rs (hunk 1 above)
Comment (overvenus) [reply components/cdc/src/endpoint.rs:908]: Confirmed, release-4.0 can filter events by `RequestId`. https://github.com/pingcap/ticdc/blob/release-4.0/cdc/kv/client.go#L1243-L1257

**Pull request merged**